File: projeto_ed_para_praticar/src/main/java/repository/Listavel.java
package repository;

//Listable
public interface Listavel {
    //Métodos principais
    void inserir(Object objeto, int posicao);   //C - insert
    void anexar(Object objeto);                 //C - append
    Object selecionar(int posicao);             //R - select
    Object[] selecionarTodos();                 //R - selectAll
    void atualizar(Object objeto, int posicao); //U - update
    Object apagar(int posicao);                 //D - delete
    void limpar();                              //D - clear

    //Métodos auxiliares
    int tamanho();
    boolean estaVazia();
    boolean estaCheia();
    String imprimir();
}

File: hash/Kotlin/hashmap/Versão2/Listavel.kt
interface Listavel {
    // Metodos principais
    fun anexar(dado: Any?)                  //C
    fun inserir(posicao: Int, dado: Any?)   //C
    fun selecionar(posicao: Int): Any?      //R
    fun selecionarTodos(): Array<Any?>      //R
    //fun selecionarProximo(): Any?           //R
    //fun selecionarAnterior(): Any?          //R   
    fun atualizar(posicao: Int, dado: Any?) //U
    fun apagar(posicao: Int): Any?          //D

    // Metodos auxiliares
    fun tamanho(): Int  
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}